### PR TITLE
BXMSPROD-957 Sonar analysis adapted to JDK11 requirement

### PR DESCRIPTION
**Pull request by Ginxo**
**Thank you for submitting this pull request**

**JIRA**: https://issues.redhat.com/browse/BXMSPROD-957

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Comment by Ginxo** [comment]
tested here https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/KIE/job/master/job/pullrequest/job/kie-soup-master.pr/15

File: .ci/sonarAnalysis.stages [hunk 1]
```diff
@@ -0,0 +1,19 @@
+def call(def propertiesFolderPath) {
+    stage('Sonar Analysis') {
+        def gitURL = env.ghprbAuthorRepoGitUrl ?: env.GIT_URL
+        def project = util.getProjectGroupName(util.getProject(gitURL))[1]
+        if(["optaplanner", "drools", "appformer", "jbpm", "drools-wb", "kie-soup", "droolsjbpm-integration", "kie-wb-common", "openshift-drools-hacep", "optaweb-employee-rostering"].contains(project))
```
Comment (mareknovotny) [inline .ci/sonarAnalysis.stages:5]: is the list of repositories complete?
I miss for instance optaweb-vehicle-routing there

Comment (Ginxo) [reply .ci/sonarAnalysis.stages:5]: @mareknovotny  they are the projects to execute sonar analysis. The list is taken from here https://github.com/kiegroup/droolsjbpm-build-bootstrap/blob/3f5746abbc7380c97878982d91cb2b2ccdc5a303/.ci/pullrequest.stages#L14
Probably @mbiarnes can clarify this better than me.

Comment (rsynek) [reply .ci/sonarAnalysis.stages:5]: @yurloc the sonarcloud analysis in optaweb-vehicle-routing is handled by travis. Do you prefer keeping it like that?

**Review by rsynek — APPROVED** [review]
Thanks for fixing the SonarCloud!

**Review by mbiarnes — APPROVED** [review]
+1

File: .ci/sonarAnalysis.stages (hunk 1 above)
Comment (Ginxo) [reply .ci/sonarAnalysis.stages:5]: @mareknovotny @mbiarnes ^

Comment (yurloc) [reply .ci/sonarAnalysis.stages:5]: I'm not against unification and I don't care whether SonarCloud analysis runs on Jenkins or Travis. What I do care about is a fast response time. So if I update a PR I want to get the analysis results within 20 minutes.

Also, the SonarCloud PR check stopped working on OptaWeb ER a few months ago. So until that is fixed I don't want to take the same path with OptaWeb VR.

Comment (mbiarnes) [reply .ci/sonarAnalysis.stages:5]: @yurloc @rsynek my opinion is to run the sonarcloud analyis by this change, so add optaweb-vehicle-routing to this list. It is more consistent I would say.

Comment (yurloc) [reply .ci/sonarAnalysis.stages:5]: I thkn you can add `optaweb-vehicle-routing` on the list if that doesn't break anything. Let me know what I need change in `optaweb-vehicle-routing/pom.xml` in order to make the analysis actually run when building on Jenkins.

Comment (Ginxo) [reply .ci/sonarAnalysis.stages]: `optaweb-vehicle-routing` added. ready to merge

**Review by mareknovotny — APPROVED** [review]

**Pull request merged**